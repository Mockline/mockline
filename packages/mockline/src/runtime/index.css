@import "tailwindcss";
@import "assets/css/animation.css";

@variant dark (&:where(.dark, .dark *));

@plugin "@tailwindcss/typography";

@theme {
  --color-neutral: var(--ui-bg);
  --color-neutral-elevated: var(--ui-bg-elevated);
  --color-border: var(--ui-border);
}

@layer base {
  :root {
    color-scheme: light;

    --ui-text-dimmed: var(--ui-color-neutral-400);
    --ui-text-muted: var(--ui-color-neutral-500);
    --ui-text-toned: var(--ui-color-neutral-600);
    --ui-text: var(--ui-color-neutral-700);
    --ui-text-highlighted: var(--ui-color-neutral-900);

    --ui-bg: var(--color-white);
    --ui-bg-elevated: var(--ui-color-neutral-100);
    --ui-bg-accented: var(--ui-color-neutral-200);
    --ui-bg-inverted: var(--ui-color-neutral-900);

    --ui-border: var(--ui-color-neutral-200);
    --ui-border-accented: var(--ui-color-neutral-300);
    --ui-border-inverted: var(--ui-color-neutral-900);

    --ui-radius: var(--radius);

    /*--ui-slate: var(--color-slate-500);
    --ui-gray: var(--color-gray-500);
    --ui-zinc: var(--color-zinc-500);
    --ui-stone: var(--color-stone-500);
    --ui-neutral: var(--color-neutral-500);
    --ui-red: var(--color-red-500);
    --ui-orange: var(--color-orange-500);
    --ui-amber: var(--color-amber-500);
    --ui-yellow: var(--color-yellow-500);
    --ui-lime: var(--color-lime-500);
    --ui-green: var(--color-green-500);
    --ui-emerald: var(--color-emerald-500);
    --ui-teal: var(--color-teal-500);
    --ui-cyan: var(--color-cyan-500);
    --ui-sky: var(--color-sky-500);
    --ui-blue: var(--color-blue-500);
    --ui-indigo: var(--color-indigo-500);
    --ui-violet: var(--color-violet-500);
    --ui-purple: var(--color-purple-500);
    --ui-fuchsia: var(--color-fuchsia-500);
    --ui-pink: var(--color-pink-500);
    --ui-rose: var(--color-rose-500);*/
  }

  .dark {
    color-scheme: dark;

    --ui-text-dimmed: var(--ui-color-neutral-500);
    --ui-text-muted: var(--ui-color-neutral-400);
    --ui-text-toned: var(--ui-color-neutral-300);
    --ui-text: var(--ui-color-neutral-200);
    --ui-text-highlighted: var(--color-white);

    --ui-bg: var(--ui-color-neutral-900);
    --ui-bg-elevated: var(--ui-color-neutral-800);
    --ui-bg-accented: var(--ui-color-neutral-700);
    --ui-bg-inverted: var(--color-white);

    --ui-border: var(--ui-color-neutral-800);
    --ui-border-accented: var(--ui-color-neutral-700);
    --ui-border-inverted: var(--color-white);
  }
}

