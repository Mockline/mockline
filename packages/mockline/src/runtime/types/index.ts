import type { Color } from '../utils/colors'

export * from '../components/elements/Button.vue'

export type ModuleOptions = {
  /**
   * Prefix for all components
   */
  prefix?: string,
  /**
   * Enable Nuxt Content
   * @default false
   */
  content?: boolean
  /**
   * The colors to use
   */
  colors?: Color[]
  /**
   * Enable Color Transitions
   * @default false
   */
  transitions?: boolean
}
