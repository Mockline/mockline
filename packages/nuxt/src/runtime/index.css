@import "@mockline/themes";
@import "./mockline.css";
