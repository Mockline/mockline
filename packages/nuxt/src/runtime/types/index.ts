// Elements
export * from '../components/Button.vue'
export * from '../components/NavMenu.vue'

// Layout
export * from '../components/App.vue'
export * from '../components/Aside.vue'
export * from '../components/Card.vue'
export * from '../components/Header.vue'
export * from '../components/Main.vue'
export * from '../components/Page.vue'
export * from '../components/PageBody.vue'
export * from '../components/PageHeader.vue'

// Settings
export * from '../components/ThemeToggle.vue'

// Overlays
export * from '../components/Toasts.vue'

// Content
export * from '../components/content/ContentNavigationTree.vue'
export * from '../components/content/ContentSurround.vue'
export * from '../components/content/ContentToc.vue'
export * from '../components/content/ContentTocLinks.vue'

// Global
export * from '../components/prose/Callout.vue'
export * from '../components/prose/CodeGroup.vue'
export * from '../components/prose/ProseCode.vue'
export * from '../components/prose/CodeButton.vue'
