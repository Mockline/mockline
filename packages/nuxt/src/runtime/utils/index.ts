export function transformSlot(slot: any, index: number): any {
  if (typeof slot.type === 'symbol') {
    return slot.children?.map(transformSlot)
  }
  return {
    label: slot.props?.filename || slot.props?.label || `${index}`,
    icon: slot.props?.icon,
    component: slot
  }
}

export * from './useComponent'
