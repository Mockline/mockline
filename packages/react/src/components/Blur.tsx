'use client'

import React, { FC } from 'react'

type BlurProps = {
  position?: 'top' | 'bottom' | 'both'
  size?: number
}

const blurLevels = [1, 2, 3, 6, 12] as const

const positions = {
  top: {
    class: 'top-0',
    gradient: 'gradient-mask-b-0',
  },
  bottom: {
    class: 'bottom-0',
    gradient: 'gradient-mask-t-0',
  },
}

export const Blur: FC<BlurProps> = ({
  position = 'top',
  size = 75,
}) => {
  return (
    <>
      {(['top', 'bottom'] as const).map((pos) => {
        if (position !== pos && position !== 'both') return null

        return (
          <div
            key={pos}
            className={`fixed w-full ${positions[pos].class}`}
            style={{ height: `${size}px` }}
          >
            {blurLevels.map((blur) => (
              <div
                key={blur}
                className={`absolute inset-0 ${positions[pos].gradient} blur-[${blur}px]`}
                style={{
                  WebkitBackdropFilter: `blur(${blur}px)`,
                  backdropFilter: `blur(${blur}px)`,
                }}
              />
            ))}
          </div>
        )
      })}
    </>
  )
}
