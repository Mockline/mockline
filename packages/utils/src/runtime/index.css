@import "tailwindcss";

@variant dark (&:where(.dark, .dark *));

@plugin "@tailwindcss/typography";
