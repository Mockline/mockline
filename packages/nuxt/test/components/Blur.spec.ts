import { describe, it, expect } from 'vitest'
import { renderSuspended } from '@nuxt/test-utils/runtime'
import Blur from '../../src/runtime/components/Blur.vue'

describe('Blur', () => {
  it('can mount component', async () => {
    const html = await renderSuspended(Blur)
    expect(html.html()).toMatchSnapshot()
  })
})
