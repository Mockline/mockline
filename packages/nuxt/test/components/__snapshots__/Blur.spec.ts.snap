// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Blur > renders <MBlur position="top" /> correctly 1`] = `
"<div data-v-109d0642="">
  <div data-v-109d0642="" class="fixed w-full top-0 h-24" style="height: 75px;">
    <div data-v-109d0642="" style="backdrop-filter: blur(1px);" class="absolute inset-0 gradient-mask-b-0 blur-[1px]"></div>
    <div data-v-109d0642="" style="backdrop-filter: blur(2px);" class="absolute inset-0 gradient-mask-b-0 blur-[2px]"></div>
    <div data-v-109d0642="" style="backdrop-filter: blur(3px);" class="absolute inset-0 gradient-mask-b-0 blur-[3px]"></div>
    <div data-v-109d0642="" style="backdrop-filter: blur(6px);" class="absolute inset-0 gradient-mask-b-0 blur-[6px]"></div>
    <div data-v-109d0642="" style="backdrop-filter: blur(12px);" class="absolute inset-0 gradient-mask-b-0 blur-[12px]"></div>
  </div>
  <!--v-if-->
</div>"
`;

exports[`Blur > renders <MBlur positoin="top" /> correctly 1`] = `
"<div data-v-109d0642="" positoin="top">
  <div data-v-109d0642="" class="fixed w-full top-0 h-24" style="height: 75px;">
    <div data-v-109d0642="" style="backdrop-filter: blur(1px);" class="absolute inset-0 gradient-mask-b-0 blur-[1px]"></div>
    <div data-v-109d0642="" style="backdrop-filter: blur(2px);" class="absolute inset-0 gradient-mask-b-0 blur-[2px]"></div>
    <div data-v-109d0642="" style="backdrop-filter: blur(3px);" class="absolute inset-0 gradient-mask-b-0 blur-[3px]"></div>
    <div data-v-109d0642="" style="backdrop-filter: blur(6px);" class="absolute inset-0 gradient-mask-b-0 blur-[6px]"></div>
    <div data-v-109d0642="" style="backdrop-filter: blur(12px);" class="absolute inset-0 gradient-mask-b-0 blur-[12px]"></div>
  </div>
  <!--v-if-->
</div>"
`;

exports[`Blur > renders both correctly 1`] = `
"<div data-v-109d0642="">
  <div data-v-109d0642="" class="fixed w-full top-0 h-24" style="height: 12px;">
    <div data-v-109d0642="" style="backdrop-filter: blur(1px);" class="absolute inset-0 gradient-mask-b-0 blur-[1px]"></div>
    <div data-v-109d0642="" style="backdrop-filter: blur(2px);" class="absolute inset-0 gradient-mask-b-0 blur-[2px]"></div>
    <div data-v-109d0642="" style="backdrop-filter: blur(3px);" class="absolute inset-0 gradient-mask-b-0 blur-[3px]"></div>
    <div data-v-109d0642="" style="backdrop-filter: blur(6px);" class="absolute inset-0 gradient-mask-b-0 blur-[6px]"></div>
    <div data-v-109d0642="" style="backdrop-filter: blur(12px);" class="absolute inset-0 gradient-mask-b-0 blur-[12px]"></div>
  </div>
  <div data-v-109d0642="" class="fixed w-full bottom-0 h-24" style="height: 12px;">
    <div data-v-109d0642="" style="backdrop-filter: blur(1px);" class="absolute inset-0 gradient-mask-t-0 blur-[1px]"></div>
    <div data-v-109d0642="" style="backdrop-filter: blur(2px);" class="absolute inset-0 gradient-mask-t-0 blur-[2px]"></div>
    <div data-v-109d0642="" style="backdrop-filter: blur(3px);" class="absolute inset-0 gradient-mask-t-0 blur-[3px]"></div>
    <div data-v-109d0642="" style="backdrop-filter: blur(6px);" class="absolute inset-0 gradient-mask-t-0 blur-[6px]"></div>
    <div data-v-109d0642="" style="backdrop-filter: blur(12px);" class="absolute inset-0 gradient-mask-t-0 blur-[12px]"></div>
  </div>
</div>"
`;

exports[`Blur > renders bottom correctly 1`] = `
"<div data-v-109d0642="">
  <!--v-if-->
  <div data-v-109d0642="" class="fixed w-full bottom-0 h-24" style="height: 12px;">
    <div data-v-109d0642="" style="backdrop-filter: blur(1px);" class="absolute inset-0 gradient-mask-t-0 blur-[1px]"></div>
    <div data-v-109d0642="" style="backdrop-filter: blur(2px);" class="absolute inset-0 gradient-mask-t-0 blur-[2px]"></div>
    <div data-v-109d0642="" style="backdrop-filter: blur(3px);" class="absolute inset-0 gradient-mask-t-0 blur-[3px]"></div>
    <div data-v-109d0642="" style="backdrop-filter: blur(6px);" class="absolute inset-0 gradient-mask-t-0 blur-[6px]"></div>
    <div data-v-109d0642="" style="backdrop-filter: blur(12px);" class="absolute inset-0 gradient-mask-t-0 blur-[12px]"></div>
  </div>
</div>"
`;

exports[`Blur > renders top correctly 1`] = `
"<div data-v-109d0642="">
  <div data-v-109d0642="" class="fixed w-full top-0 h-24" style="height: 12px;">
    <div data-v-109d0642="" style="backdrop-filter: blur(1px);" class="absolute inset-0 gradient-mask-b-0 blur-[1px]"></div>
    <div data-v-109d0642="" style="backdrop-filter: blur(2px);" class="absolute inset-0 gradient-mask-b-0 blur-[2px]"></div>
    <div data-v-109d0642="" style="backdrop-filter: blur(3px);" class="absolute inset-0 gradient-mask-b-0 blur-[3px]"></div>
    <div data-v-109d0642="" style="backdrop-filter: blur(6px);" class="absolute inset-0 gradient-mask-b-0 blur-[6px]"></div>
    <div data-v-109d0642="" style="backdrop-filter: blur(12px);" class="absolute inset-0 gradient-mask-b-0 blur-[12px]"></div>
  </div>
  <!--v-if-->
</div>"
`;
